database/merkledb/node.go

```diff
@@ -0,0 +1,30 @@
+package merkledb
+
+import (
+	"github.com/ava-labs/avalanchego/utils/hashing"
+)
+
+type Node interface {
+	GetChild(key Key) (Node, error)
+	GetNextNode(prefix Key, start Key, key Key) (Node, error)
+	Insert(key Key, value []byte) error
+	Delete(key Key) error
+	SetChild(node Node) error
+	SetParent(b Node)
+	SetPersistence(p Persistence)
+	Value() []byte
+	Hash(key Key, hash []byte) error
+	GetHash() []byte
+	GetPreviousHash() []byte
+	References(change int32) int32
+	Key() Key
+	GetChildrenHashes() [][]byte
+	GetReHash() []byte
+	Clear() error
+	Print()
+}
+
+func Hash(bs ...[]byte) []byte {
+	hash := hashing.ByteArraysToHash256Array(bs...)
+	return hash[:]
+}
```

Comment:
This function should be removed.

Reply:
It's slicing the `[32]byte` into `[]byte`

Changing everywhere from `b.StoredHash = Hash(hashSet...)` to 
 ```
 storedHash := hashing.ByteArraysToHash256Array(hashSet...)
 b.StoredHash = storedHash[:]
 ``` 
Is kinda wonky.

Should we store [32]byte hashes in the Node instead of []byte ?